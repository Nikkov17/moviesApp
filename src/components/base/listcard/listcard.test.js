import React from 'react';
import Listcard from './listcard.component';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

describe('Listcard', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  describe('snapshot', () => {
    it('check snaphot',() => {
      const component = shallow( <Listcard item={{
          title: 'check title',
          description: 'check description',
          img: 'check img'
      }}/>);
      expect(component).toMatchSnapshot();
    })
  })
});