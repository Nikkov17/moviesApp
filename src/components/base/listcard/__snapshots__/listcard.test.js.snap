// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listcard snapshot check snaphot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListCard
    item={
      Object {
        "description": "check description",
        "img": "check img",
        "title": "check title",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="movie poster"
          className="img"
          src="check img"
        />,
        <p
          className="card-text movie-title"
        >
          check title
        </p>,
        <p
          className="card-text movie-description"
        >
          check description
        </p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "className": "img",
          "src": "check img",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "check title",
          "className": "card-text movie-title",
        },
        "ref": null,
        "rendered": "check title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "check description",
          "className": "card-text movie-description",
        },
        "ref": null,
        "rendered": "check description",
        "type": "p",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie poster"
            className="img"
            src="check img"
          />,
          <p
            className="card-text movie-title"
          >
            check title
          </p>,
          <p
            className="card-text movie-description"
          >
            check description
          </p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "className": "img",
            "src": "check img",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "check title",
            "className": "card-text movie-title",
          },
          "ref": null,
          "rendered": "check title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "check description",
            "className": "card-text movie-description",
          },
          "ref": null,
          "rendered": "check description",
          "type": "p",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
